src/go/configgenerator/listener_generator.go

```diff
@@ -50,8 +50,87 @@ const (
 	statPrefix = "ingress_http"
 )
 
+// MakeListeners provides dynamic listeners for Envoy
+func MakeListeners(serviceInfo *sc.ServiceInfo) ([]*v2pb.Listener, error) {
+	var listeners []*v2pb.Listener
+
+	port := serviceInfo.Options.ListenerPort
+	isSsl := false
+	if serviceInfo.Options.SslPort != 0 {
+		port = serviceInfo.Options.SslPort
+		isSsl = true
+	}
+
+	listener, err := makeListener(serviceInfo, port, isSsl)
+	if err != nil {
+		return nil, err
+	}
+	listeners = append(listeners, listener)
+
+	redirect_listener, err := makeRedirectListener(serviceInfo)
+	if err != nil {
+		return nil, err
+	}
+	if serviceInfo.Options.SslPort != 0 {
+		listeners = append(listeners, redirect_listener)
+	}
+
+	return listeners, nil
+}
+
+func makeRedirectListener(serviceInfo *sc.ServiceInfo) (*v2pb.Listener, error) {
+	httpFilters := []*hcmpb.HttpFilter{}
+	routerFilter := makeRouterFilter(serviceInfo.Options)
+	httpFilters = append(httpFilters, routerFilter)
+	route, err := MakeRedirectRouteConfig(serviceInfo)
+	if err != nil {
+		return nil, fmt.Errorf("MakeRedirectRouteConfig got err: %s", err)
+	}
+
+	httpConMgr := &hcmpb.HttpConnectionManager{
+		CodecType:  hcmpb.HttpConnectionManager_AUTO,
+		StatPrefix: statPrefix,
+		RouteSpecifier: &hcmpb.HttpConnectionManager_RouteConfig{
+			RouteConfig: route,
+		},
+	}
+
+	jsonStr, _ := util.ProtoToJson(httpConMgr)
+	glog.Infof("adding Http Connection Manager config: %v", jsonStr)
```

Comment:
how about move this glog after line 100 to include filter chain data

Reply:
Done